dts/bindings/modem/quectel,bg96.yaml

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2022 Trackunit
+# SPDX-License-Identifier: Apache-2.0
+
+description: Quectel BG96 modem
+
+compatible: "quectel,bg96"
+
+include: [base.yaml, uart-device.yaml]
+
+properties:
+    label:
+      required: true
+
+    power-gpios:
+      type: phandles
+      required: true
+
+    reset-gpios:
+      type: phandles
+      required: false
+
+    powerdown-gpios:
+      type: phandles
+      required: false
+
+    nmea-uart:
+      type: path
+      required: false
+      description: |
+        Used for NMEA frames from GNSS if included,
+        otherwise,the location is read from the modem
+        using AT commands.
+
+    apn-name:
+      type: string
+      required: false
+      description: APN name for network connection context
+
+    username:
+      type: string
+      required: false
+      description: Username for network connection context
+
+    password:
+      type: string
+      required: false
+      description: Password for network connection context
```

Comment:
Without commenting on the meat of the PR: a friendly but unambiguous **no way** to these properties :). Not now, not ever, not on my watch :). CC @d3zd3z 

- While this isn't strictly speaking a security hole since the password property will not automatically be stored in the firmware, it is just too tempting a target. Someone somewhere will save into ROM "just for now"; I can't in good conscience encourage this antipattern.
- IMO this is a pain to manage from a provisioning standpoint. Are we really going to expect every single device to get its own firmware build just to pass in an overlay for the unique password (the password **is** unique per device, right? :slightly_smiling_face: )
- This is also a pain from a flexibility standpoint. This might work for you, but other people are going to want to configure their software differently, e.g. getting the password from a special piece of on-chip flash with anti-readout protections, or from an EEPROM, or ...

TL;DR this is the sort of software configuration that doesn't belong in DT, and it is a security smell to boot.

Reply:
> This is currently done using KConfig, which to my knowledge has the exact same issues as you mention here :)

The presence of bad behavior elsewhere is not a justification for introducing it into a devicetree binding.

> The username and password are like WIFI passwords, tied to the mobile network provider. All units using the same provider will use the same username/password, if required.

Then it doesn't seem like this is per-instance modem configuration, but rather per-mobile-network information, which also argues against having them here.

> most APNs and their accompanying passwords are public information https://www.hw-group.com/support/international-apn-settings-for-mobile-broadband-network-operators

OK, thanks for explaining that. That does indeed change the security aspect of this in most cases, I guess. However, "most" is not "all".

> Usually, mass manufactured devices have soldered on ESIM chips which have a fixed APN/Username/Password, so it can be tied directly with the device through firmware.

Right, which is exactly my point about flexibility, which is basically that putting this in DT doesn't really make sense if people are going to want to do it in other ways, and in fact that seems to be the case here, which is not surprising.

> which require the ability to change SIM card and APN during runtime,

In general, devicetree describes the hardware and its **boot time** configuration. Just because something can change at runtime is not -- in general -- a reason to remove its boot time value from the DT. However again I still don't think this should go here.

> but for now, this is how it is done.

My -1 remains, sorry

Reply:
Well, can you propose a better solution to the issue? Keep in mind that two modems can and would likely use two different SIM cards from different providers for redundancy.